stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/avatar/AvatarStyle.kt

```diff
@@ -0,0 +1,67 @@
+package io.getstream.chat.android.ui.avatar
+
+import android.content.Context
+import android.content.res.TypedArray
+import android.graphics.Color
+import android.graphics.Typeface
+import android.util.AttributeSet
+import com.getstream.sdk.chat.style.TextStyle
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.utils.getDimension
+
+public class AvatarStyle(context: Context, attrs: AttributeSet?) {
+    public val avatarWidth: Int
+    public val avatarHeight: Int
+    public val avatarBorderWidth: Int
+    public val avatarBorderColor: Int
+    public val avatarInitialText: TextStyle
+    public val onlineIndicatorEnabled: Boolean
+
+    init {
+        val a: TypedArray = context.obtainStyledAttributes(
+            attrs,
+            R.styleable.AvatarView,
+            0,
+            0
+        )
+
+        avatarWidth = a.getDimensionPixelSize(
+            R.styleable.AvatarView_streamAvatarWidth,
+            context.getDimension(R.dimen.stream_message_avatar_width)
+        )
+        avatarHeight = a.getDimensionPixelSize(
+            R.styleable.AvatarView_streamAvatarHeight,
+            context.getDimension(R.dimen.stream_message_avatar_height)
+        )
+        avatarBorderWidth = a.getDimensionPixelSize(
+            R.styleable.AvatarView_streamAvatarBorderWidth,
+            context.getDimension(R.dimen.stream_channel_avatar_border_width)
+        )
+        avatarBorderColor = a.getColor(
+            R.styleable.AvatarView_streamAvatarBorderColor,
+            Color.WHITE
+        )
+        avatarInitialText = TextStyle.Builder(a)
+            .size(
+                R.styleable.AvatarView_streamAvatarTextSize,
+                context.getDimension(R.dimen.stream_channel_initials)
+            )
+            .color(
+                R.styleable.AvatarView_streamAvatarTextColor,
+                Color.WHITE
+            )
+            .font(
+                R.styleable.AvatarView_streamAvatarTextFontAssets,
+                R.styleable.AvatarView_streamAvatarTextFont
+            )
+            .style(
+                R.styleable.AvatarView_streamAvatarTextStyle,
+                Typeface.BOLD
+            )
+            .build()
+        onlineIndicatorEnabled = a.getBoolean(
+            R.styleable.AvatarView_streamAvatarOnlineIndicatorEnabled,
+            false
+        )
+    }
```

Comment:
The TypedArray should be recycled when no longer needed. I think that we can wrap everything with `.use` to do that automatically

Reply:
I've just added add `a.recycle()` in the end. Don't want to wrap into`.use` as we will need to use `lateinit` fields, some of them like `Int` are considered primitives and need to be initialized with some default value.

Reply:
Good call, better than going `lateinit` with everything. A real shame that AndroidX methods don't use Kotlin contracts yet, it would solve this issue really easily.

Although... We could add our own `use` method that supports contracts if we wanted - might be worth doing, since we'll do a bunch of attribute parsing like this. Would be pretty much like this:

```kotlin
internal inline fun TypedArray.use(block: (TypedArray) -> Unit) {
    contract { 
        callsInPlace(block, InvocationKind.EXACTLY_ONCE)
    }
    block(this)
    recycle()
}
```

Reply:
Done